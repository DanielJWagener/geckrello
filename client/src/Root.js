// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
