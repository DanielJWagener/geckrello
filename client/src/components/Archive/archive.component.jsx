import React from "react";
import { connect } from "react-redux";

import ArchiveItem from "../ArchiveItem/archive-item.component";

class Archive extends React.Component {
  state = { mode: "lists" };

  toggleMode = () => {
    if (this.state.mode === "lists") {
      this.setState({ mode: "cards" });
    } else {
      this.setState({ mode: "lists" });
    }
  };

  archivedCardsArray = () =>
    this.props.cards
      .filter(card => card.archived)
      .map(card => (
        <ArchiveItem
          key={card._id}
          itemId={card._id}
          itemTitle={card.title}
          itemType="card"
        />
      ));

  archivedListsArray = () =>
    this.props.lists
      .filter(list => list.archived)
      .map(list => (
        <ArchiveItem
          key={list._id}
          itemId={list._id}
          itemTitle={list.title}
          itemType="list"
        />
      ));

  render() {
    return (
      <div>
        <div className="archive-toggle">
          <button
            className={`archive-toggle__button archive-toggle__button${
              this.state.mode === "lists" ? "--active" : "--inactive"
            }`}
            onClick={this.toggleMode}
          >
            Lists
          </button>
          <button
            className={`archive-toggle__button archive-toggle__button${
              this.state.mode === "cards" ? "--active" : "--inactive"
            }`}
            onClick={this.toggleMode}
          >
            Cards
          </button>
        </div>
        <div className="archive__list">
          {this.state.mode === "lists"
            ? this.archivedListsArray()
            : this.archivedCardsArray()}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return { lists: state.lists, cards: state.cards };
};

export default connect(mapStateToProps)(Archive);
