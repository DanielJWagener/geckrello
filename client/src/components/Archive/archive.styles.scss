@import "../../sass/abstracts/_variables.scss";

.archive {
  &-toggle {
    display: flex;
    justify-content: center;

    &__button {
      width: 25%;
      font-size: 2rem;
      color: white;
      padding: 0.5rem 1rem 0.5rem 1rem;
      border-radius: 5px;
      border: none;
      margin: 0.5rem;

      &:focus {
        outline: none;
      }

      &--inactive {
        background-color: #444;
        cursor: pointer;
      }

      &--active {
        background-color: blue;
      }
    }
  }

  &__list {
    margin-top: 2rem;
  }

  &__item {
    display: flex;
    justify-content: space-around;
    align-items: flex-center;
    margin-bottom: 2rem;

    &--title {
      font-size: 2rem;
      padding: 1em 4rem 1rem 4rem;
      text-align: center;
      background-color: #ddd;
      border-radius: 5px;
      width: 50%;
    }

    &--restore {
      font-size: 2rem;
      padding: 0.5rem 1rem 0.5rem 1rem;
      text-align: center;
      background-color: blue;
      color: white;
      border-radius: 5px;
      cursor: pointer;
      width: 25%;
    }
  }
}
