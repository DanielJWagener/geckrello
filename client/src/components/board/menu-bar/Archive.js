import React from "react";
import ArchiveInner from "./ArchiveInner";

const Archive = props => {
  if (props.hidden) {
    return (
      <div className="sidebar-menu__interface sidebar-menu__interface--hidden">
        <ArchiveInner toggleArchive={props.toggleArchive} />
      </div>
    );
  } else {
    return (
      <div className="sidebar-menu__interface sidebar-menu__interface--visible">
        <ArchiveInner toggleArchive={props.toggleArchive} />
      </div>
    );
  }
};

export default Archive;
