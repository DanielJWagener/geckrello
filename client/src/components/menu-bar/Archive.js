import React from "react";
import ArchiveInner from "./ArchiveInner";

const Archive = ({ hidden, toggleArchive }) => (
  <div
    className={`sidebar-menu__interface sidebar-menu__interface--${
      hidden ? "hidden" : "visible"
    }`}
  >
    <ArchiveInner toggleArchive={toggleArchive} />
  </div>
);

export default Archive;
