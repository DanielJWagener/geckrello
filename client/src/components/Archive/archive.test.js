import { shallow, mount } from "enzyme";
import React from "react";
import Archive from "./archive.component";
import { Archive as ArchivePure } from "./archive.component";
import Root from "../../Root";

const mockProps = {
  cards: [
    { _id: "1", title: "Card 1", archived: true },
    { _id: "2", title: "Card 2", archived: false },
  ],
  lists: [
    { _id: "1", title: "List 1", archived: true },
    { _id: "2", title: "List 2", archived: false },
  ],
};

it("renders Archive component", () => {
  expect(shallow(<ArchivePure {...mockProps} />).debug()).toMatchSnapshot();
});

describe("Archive functionality", () => {
  let wrapped;

  beforeEach(() => {
    wrapped = mount(
      <Root>
        <Archive />
      </Root>
    );
  });

  afterEach(() => {
    wrapped.unmount();
  });

  it("allows users to toggle between cards and lists", () => {
    const activeButton = ".archive-toggle__button--active";
    const inactiveButton = ".archive-toggle__button--inactive";
    const activeInitial = wrapped.find(activeButton).text();
    const inactiveInitial = wrapped.find(inactiveButton).text();

    // Both the inactive and active buttons should work. First the active:
    wrapped.find(activeButton).simulate("click");
    expect(wrapped.find(activeButton).text()).toEqual(inactiveInitial);
    wrapped.find(activeButton).simulate("click");
    expect(wrapped.find(activeButton).text()).toEqual(activeInitial);

    // Now the inactive:
    wrapped.find(inactiveButton).simulate("click");
    expect(wrapped.find(activeButton).text()).toEqual(inactiveInitial);
    wrapped.find(inactiveButton).simulate("click");
    expect(wrapped.find(activeButton).text()).toEqual(activeInitial);
  });

  it("displays only archived cards and lists", () => {
    // We export the Archive component before it gets passed into connect()
    // This way, we can call instance() on the component without a Provider getting in the way
    const wrapped2 = shallow(<ArchivePure {...mockProps} />);

    expect(wrapped2.instance().archivedItemsArray("cards").length).toEqual(1);
    expect(wrapped2.instance().archivedItemsArray("lists").length).toEqual(1);
  });
});
