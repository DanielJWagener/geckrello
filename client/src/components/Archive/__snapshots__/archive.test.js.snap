// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Archive component 1`] = `
"<div>
  <div className=\\"archive-toggle\\">
    <button className=\\"archive-toggle__button archive-toggle__button--active\\" onClick={[Function]}>
      Lists
    </button>
    <button className=\\"archive-toggle__button archive-toggle__button--inactive\\" onClick={[Function]}>
      Cards
    </button>
  </div>
  <div className=\\"archive__list\\">
    <Connect(ArchiveItem) itemId=\\"1\\" itemTitle=\\"List 1\\" itemType=\\"list\\" />
    <Connect(ArchiveItem) itemId=\\"2\\" itemTitle=\\"List 2\\" itemType=\\"list\\" />
  </div>
</div>"
`;

exports[`renders ArchiveItem component 1`] = `
"<div className=\\"archive__item\\">
  <div className=\\"archive__item--title\\" />
  <button className=\\"archive__item--restore\\" onClick={[Function: restore]}>
    Restore
  </button>
</div>"
`;
